pytest/conftest.py

```diff
@@ -0,0 +1,219 @@
+"""This file sets up custom plugins.
+
+https://docs.pytest.org/en/stable/writing_plugins.html
+
+"""
+from __future__ import annotations
+
+import typing
+from pathlib import Path
+
+import schema
+
+import lisa
+import playbook
+import pytest
+
+# TODO: Use importlib instead
+from azure import Azure
+from target import Target
+
+if typing.TYPE_CHECKING:
+    from typing import Any, Dict, Iterator, List, Optional
+
+    from _pytest.config import Config
+    from _pytest.config.argparsing import Parser
+    from _pytest.fixtures import FixtureRequest
+    from _pytest.python import Metafunc
+
+    from pytest import Item, Session
+
+
+LISA = pytest.mark.lisa
+LINUX_SCRIPTS = Path("../Testscripts/Linux")
+
+
+@pytest.fixture(scope="session")
+def pool(request: FixtureRequest) -> Iterator[List[Target]]:
+    """This fixture tracks all deployed target resources."""
+    targets: List[Target] = []
+    yield targets
+    for t in targets:
```

Comment:
When the targets get deleted? It needs to be deleted, when no more test case need it to save cost.

Reply:
When they're no longer in use!

Reply:
My plan is for high-cost VMs (as determined by an API check that @lpereira found) that we de-allocate when they're no longer the currently-in-use target, and then delete at the end with the rest.